perma_web/perma/tasks.py

```diff
@@ -265,35 +386,316 @@ def makefile(self, *args, **kwargs):
 
     return requests_response
 
-def run_in_frames(browser, func, output_collector=None):
-    # setup
-    browser.implicitly_wait(0)
-    if output_collector is None:
-        output_collector = []
 
-    run_in_frames_recursive(browser, func, output_collector)
+### CAPTURE COMPONENTS ###
+
+# x-robots headers
+
+def xrobots_blacklists_perma(robots_directives):
+    darchive = False
+    if robots_directives:
+        for directive in robots_directives.split(";"):
+            parsed = directive.lower().split(":")
+            # respect tags that target all crawlers (no user-agent specified)
+            if len(parsed) == 1:
+                if "noarchive" in parsed:
+                    darchive = True
+            # look for perma user-agent
+            elif len(parsed) == 2:
+                if parsed[0] == "perma" and "noarchive" in parsed[1]:
+                    darchive = True
+            # if the directive is poorly formed, do our best
+            else:
+                if "perma" in directive and "noarchive" in directive:
+                    darchive = True
+    return darchive
 
-    # reset
-    browser.implicitly_wait(ELEMENT_DISCOVERY_TIMEOUT)
-    browser.switch_to.default_content()
+# meta tags
 
-    return output_collector
+def meta_thread(browser=None, link=None):
```

Comment:
Having a late night memory that one can only pass kwargs into the thread-starting function, not positional args: did this so they'd be named. I will verify if that's right, or a relic of late night reasoning.